src/main/java/org/fcrepo/upgrade/utils/F47ToF5UpgradeManager.java

```diff
@@ -340,37 +441,39 @@ private TemporalAccessor resolveMementoTimestamp(final Path path) {
         var metadataPath = path;
         if (!path.toString().endsWith(TURTLE_EXTENSION)) {
             final var metadataPathStr = metadataPath.toString();
-            final var index = metadataPathStr.lastIndexOf(".");
-            final var newMetadataPathStr = metadataPathStr
-                                               .substring(0,
-                                                          index) + File.separator + FCR_METADATA_PATH_SEGMENT + TURTLE_EXTENSION;
+            final var newMetadataPathStr = removeExtension(metadataPathStr) + File.separator + FCR_METADATA_PATH_SEGMENT + TURTLE_EXTENSION;
             metadataPath = Path.of(newMetadataPathStr);
         }
 
         //resolve the subject associated with the resource
-        final Model model = ModelFactory.createDefaultModel();
-        try (final FileInputStream is = new FileInputStream(metadataPath.toFile())) {
-            RDFDataMgr.read(model, is, Lang.TTL);
-        } catch (IOException ex) {
-            throw new RuntimeException(ex);
-        }
+        final Model model = createModelFromFile(metadataPath);
         final var subject = model.listSubjects().nextResource();
 
         //resolve the versions file
         final var versionsContainer = resolveVersionsContainer(path);
         //read the versions container in order to resolve the version timestamp
-        final Model versionsContainerModel = ModelFactory.createDefaultModel();
-        try (final FileInputStream is = new FileInputStream(versionsContainer.toFile())) {
-            RDFDataMgr.read(versionsContainerModel, is, Lang.TTL);
-        } catch (IOException ex) {
-            throw new RuntimeException(ex);
-        }
+        final Model versionsContainerModel = createModelFromFile(versionsContainer);
         final var iso8601Timestamp = versionsContainerModel.listObjectsOfProperty(subject, FEDORA_CREATED_DATE)
                                                            .next().asLiteral().getString();
         //create memento id based on RFC 8601 timestamp
         return ISO_DATE_TIME_FORMATTER.parse(iso8601Timestamp);
     }
 
+    private String removeExtension(String path) {
```

Comment:
Probably less fragile to use [FilenameUtils](https://commons.apache.org/proper/commons-io/apidocs/org/apache/commons/io/FilenameUtils.html#removeExtension-java.lang.String-).